pkg/agents/team4/finalAgent/agent.go

```diff
@@ -36,153 +49,202 @@ type CustomAgentEvo struct {
-			// }
-		} else if a.HP() <= a.HealthInfo().WeakLevel+2*healthLevelSeparation { //normal
-			a.params.healthStatus = 2
-			// if dayPass {
-			// 	a.params.traumaScaleFactor = math.Max(0, a.params.traumaScaleFactor-0.02)
-			// }
-		} else { //strong
-			a.params.healthStatus = 3
-			// if dayPass {
-			// 	a.params.traumaScaleFactor = math.Max(0, a.params.traumaScaleFactor-0.03)
-			// }
-		}
+	if a.hasReshuffled() {
+		//update global trust if agents don't reply to messages
+		timesBlanked := len(a.params.sentMessages) - a.params.receivedMessagesCount
+		a.addToGlobalTrust(-a.params.trustCoefficients[0] * float64(timesBlanked))
+
+		//reset the arrays and maps when we have reshuffled
+		a.params.agentResponseMemory = make(map[uuid.UUID][]int)
+		a.params.msgToSendBuffer = []messages.Message{}
+		a.params.sentMessages = []messages.Message{}
+		a.params.responseMessages = []messages.Message{}
+		a.params.requestLeaveFoodMessages = []messages.Message{}
+		a.params.otherMessageBuffer = []messages.Message{}
+		a.params.receivedMessagesCount = 0
 	}
 
-	if a.params.globalTrust < a.params.globalTrustLimit {
-		a.params.currentPersonality = "selfish"
-	} else {
-		a.params.currentPersonality = "selfless"
+	if a.hasDayPassed() {
+		//generates messages to send per day altogether and updates params for each new day
+		a.generateMessagesToSend()
+		a.updateLastTimeFoodSeen()
+		a.updateCraving()
+	}
+
+	//update last food amount seen on the platform
+	if food.FoodType(a.CurrPlatFood()) != a.params.lastPlatFood && a.PlatformOnFloor() {
+		a.params.lastPlatFood = a.CurrPlatFood()
+	}
+
+	// receive message
+	a.getMessage()
+
+	a.params.healthStatus = getHealthStatus(a.HealthInfo(), healthLevelSeparation, a.HP())
+
+	//call the relevant message handler
+	a.callHandleMessage()
+
+	a.sendingMessage()
+
+	a.checkAgentResponseMemory() // check if we received the Hp message and intended food message and update trust score
+
+	a.setPersonality()
+
+	// if platform is on floor or floor below match agent response to sent message
+	if a.PlatformOnFloor() || (!a.PlatformOnFloor() && a.CurrPlatFood() != -1) {
+		a.verifyResponses()
 	}
 
-	foodEaten := food.FoodType(0)
+	//----------------------EATING FOOD -------------------------//
+
+	var foodEaten food.FoodType
 	var err error
-	calculatedAmountToEat := food.FoodType(0)
 
-	if (a.Age()-a.params.ageLastEaten) >= a.params.daysToWait[a.params.currentPersonality][a.params.healthStatus] || a.params.healthStatus == 0 {
-		a.params.locked = false
-		calculatedAmountToEat = food.FoodType(a.params.traumaScaleFactor * float64(a.params.foodToEat[a.params.currentPersonality][a.params.healthStatus]))
-		foodEaten, err = a.TakeFood(food.FoodType(calculatedAmountToEat))
-		a.params.ageLastEaten = a.Age()
+	if rand.Intn(100) >= (a.params.waitProbability[a.params.currentPersonality][a.params.healthStatus]-a.params.craving) && !a.HasEaten() && a.PlatformOnFloor() {
```

Comment:
I have no idea what this is comparing - you should probably put the `waitProbability` thing into a separate variable